sleepyq/__init__.py

```diff
@@ -36,66 +36,208 @@ def __init__(self, data):
-
                 setattr(bed, side, status)
         return beds
 
 
     def bed_family_status(self):
-        url = 'https://api.sleepiq.sleepnumber.com/rest/bed/familyStatus'
-        r = self._session.get(url)
-        r.raise_for_status()
-
+        url = '/bed/familyStatus'
+        r=self.__makeRequest(url)
         statuses = [FamilyStatus(status) for status in r.json()['beds']]
         return statuses
+
+    def set_light(self, bedId, light, setting):
+        #
+        # light 1-4
+        ### 1=Right Night Stand
+        ### 2=Left Night Stand
+        ### 3=Right Night Light
+        ### 4=Left Night Light
+        # setting 0=off, 1=on
+        #
+        if 1 <= light <= 4 and setting in (0, 1):
+            url = '/bed/'+bedId+'/foundation/outlet'
+            data = {'outletId': light, 'setting': setting}
+            r=self.__makeRequest(url, "put", data)
+            return True
+        else:
+            return False
+
+    def get_light(self, bedId, light):
+        #
+        # same light numbering as set_light
+        #
+        if 1 <= light <= 4:
+            url = '/bed/'+bedId+'/foundation/outlet'
+            self._session.params['outletId'] = light
+            r=self.__makeRequest(url)
+            del self._session.params['outletId']
+            light = Light(r.json())
+            return light
+        else:
+            return False
+
+    def preset(self, bedId, preset, side, speed):
+        #
+        # preset 1-6
+        ### 1=fav?
+        ### 2=read
+        ### 3=watch tv
+        ### 4=flat
+        ### 5=zero g
+        ### 6=snore
+        # side "R" or "L"
+        # Speed 0=fast, 1=slow
+        #
+        if side.lower() in ('r', 'right'):
+            side = "R"
+        elif side.lower() in ('l', 'left'):
+            side = "L"
+        else:
+            return False
+        if 1 <= preset <= 6 and speed in (0, 1):
+            url = '/bed/'+bedId+'/foundation/preset'
+            data = {'preset':preset,'side':side,'speed':speed}
+            r=self.__makeRequest(url, "put", data)
+            return True
+        else:
+            return False
+
+    def set_sleepnumber(self, bedId, side, setting):
+        #
+        # side "R" or "L"
+        # setting 0-100 (increments of 5)
+        #
+        if setting%5 != 0:
+            return False
+        if side.lower() in ('r', 'right'):
+            side = "R"
+        elif side.lower() in ('l', 'left'):
+            side = "L"
+        else:
+            return False
+        url = '/bed/'+bedId+'/sleepNumber'
+        data = {'bed': bedId, 'side': side, "sleepNumber":setting}
+        self._session.params['side']=side
+        r=self.__makeRequest(url, "put", data)
+        del self._session.params['side']
+        return True
+
+    def set_favsleepnumber(self, bedId, side, setting):
+        #
+        # side "R" or "L"
+        # setting 0-100 (increments of 5)
+        #
+        if setting%5 != 0:
+            return False
```

Comment:
An error for this with what failed would help inform the user what went wrong. Another option could be rounding to the nearest 5.

It also isn't catching other values, ie negative or more than 100.

Reply:
Changed to raise ValueError if too small or too large.  Also set the SleepNumber to round to the nearest multiple of 5.  Should it raise an error if it has to be rounded?